flocker/node/test/test_deploy.py

```diff
@@ -666,8 +669,7 @@ def test_does_not_exist(self):
 # XXX For now we require volume names match application names,
 # see https://github.com/ClusterHQ/flocker/issues/49
 DATASET_ID = unicode(uuid4())
-DATASET = Dataset(dataset_id=DATASET_ID,
-                  metadata=pmap({u"name": APPLICATION_WITH_VOLUME_NAME}))
+DATASET = Dataset(dataset_id=DATASET_ID)
```

Comment:
That metadata wasn't there to cause any particular code paths to be exercised?  And it seems like the comment is out of date now (and perhaps has been for a while - since we switched to UUIDs for dataset ids, I suppose?).  The dataset no longer has a matching name nor a matching "name" metadata item and it seems the `ApplicationNodeDeployer` logic also doesn't care about either of these things (not sure when that changed).  If this is right, delete the comment.